antarest/study/business/link_management.py

```diff
@@ -11,123 +11,35 @@
 # This file is part of the Antares project.
 
 import typing as t
-from typing import Any
 
-from antares.study.version import StudyVersion
-
-from antarest.core.exceptions import LinkNotFound
 from antarest.core.model import JSON
-from antarest.study.business.model.link_model import LinkBaseDTO, LinkDTO, LinkInternal
-from antarest.study.business.utils import execute_or_add_commands
+from antarest.study.business.model.link_model import LinkBaseDTO, LinkDTO
+from antarest.study.dao.dao_factory import DAOFactory
 from antarest.study.model import RawStudy, Study
-from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
-from antarest.study.storage.storage_service import StudyStorageService
-from antarest.study.storage.variantstudy.model.command.create_link import CreateLink
-from antarest.study.storage.variantstudy.model.command.remove_link import RemoveLink
-from antarest.study.storage.variantstudy.model.command.update_link import UpdateLink
 
 
 class LinkManager:
-    def __init__(self, storage_service: StudyStorageService) -> None:
-        self.storage_service = storage_service
+    def __init__(self, dao_factory: DAOFactory) -> None:
+        self.composite_dao = dao_factory.create_link_dao()
```

Comment:
It's important that we keep an abstract class DAO here, because the manager layer should not know what implementation of DAO it's using.

Typically, for unit tests we could use an in-memory DAO implementation which does not know anything about caching or the study file format or a databse